packages/adblocker-content/adblocker.ts

```diff
@@ -81,6 +94,13 @@ export function extractFeaturesFromDOM(roots: Element[]): {
         '[id]:not(html):not(body),[class]:not(html):not(body),[href]:not(html):not(body)',
       ),
     ]) {
+      // Check if this object belongs to seenElements and skip if we already did the job on this object.
```

Comment:
Updated to

```
// If one of root belongs to another root which is parent node of the one, querySelectorAll can return duplicates.
```

Reply:
Can you explain again how that's possible? I thought MutationObserver with subtree: false would not return nested elements.

Reply:
Still, mutation observer callback can be called multiple times with both elements mentioned in the above comment. It can call callback with the root element for the first time, and later with child element with other mutation records. In this case, using `querySelectorAll` can have some overlaps in their children.